docker/README.md

```diff
@@ -0,0 +1,168 @@
+
+As an example, the **KubeStellar** container images available at https://quay.io/repository/kubestellar/kubestellar?tab=tags have been built with commands like:
+
+```bash
+make buildx IMG=quay.io/kubestellar/kubestellar KUBESTELLAR_VERSION=v0.4.0
+```
+
+## Run the container and access KubeStellar from the host OS
+
+Spin up a local or remote container image with the command:
+
+```bash
+make run
+export KUBECONFIG=${HOME}/.kcp/admin.kubeconfig
+export PATH=$PATH:${HOME}/kcp:${HOME}/kubestellar/bin
+```
+
+The following arguments can be used:
+
+- `IMG=[<registry>/<organization>/]<image>` select the container image, the default value is the local `kubestellar` image. Use `IMG=quay.io/kubestellar/kubestellar` to pull the **stable** container image from our [quay.io registry](https://quay.io/repository/kubestellar/kubestellar?tab=tags).
+- `TAG=<tag>`, select the image tag, the default value is the **stable** version of **KubeStellar** as specified in https://github.com/kubestellar/kubestellar/blob/main/VERSION.
+- `NAME=<container-name>` sets the name of the container, the default value is `kubestellar`
+- `BASEPATH=<path>` set the base path where the **KUBECONFIG** and executable will be made available. By default, `~` is used. Note that this parameter will impact the exact path of the `export` commands above, which are provided for the default case. The `make run` command will print out the correct export commands to be used.
+
+The command above will:
+
+- share a `${BASEPATH}/.kcp` folder containing the `KUBECONFIG` file
+- share a `${BASEPATH}/.kubestellar-logs` folder containing the log files of **kcp** and **KubeStellar** controllers
+- create a `${BASEPATH}/kcp` folder containing a copy of **kcp** plugins for the host OS architecture
+- create a `${BASEPATH}/kubestellar` folder containing a copy of **KubeStellar** executables for the host OS architecture
+
+At this point, we can check basic **KubeStellar** functionality with the command:
+
+```bash
+kubectl ws root
+kubectl ws tree
+```
+
+which should return something like:
+
+```text
+.
+└── root
+    ├── compute
+    └── espw
+```
+
+At this point, one can follow the instructions and examples in **KubeStellar** [Quickstart](https://docs.kubestellar.io/release-0.3/Getting-Started/quickstart/).
+
+## Checking the container logs
+
+The running container logs can be easily displayed using the command:
+
+```bash
+make logs [NAME=<name>]
+```
+
+The `NAME` argument above is only necessary if a container name was specified with a similar argument in the `make run` command.
+
+## Login into the running container to use **KubeStellar** from inside the container
+
+Login using the command:
+
+```bash
+make exec [NAME=<name>]
+```
+
+Since `kubectl` is available inside the container, at this point **KubeStellar** can be used as before.
+
+The `NAME` argument above is only necessary if a container name was specified with a similar argument in the `make run` command.
+
+## Teardown and cleanup

```

Comment:
I think it would be better if killing the container were separate from deleting the image.  That separation is the normal pattern in the container world.

Reply:
@francostellari I agree with @MikeSpreitzer on this comment.  Let's make teardown a 2-command process.  One to kill (force stop and remove the running container) and the other to delete (remove container image from local registry and delete created folders).  That seems fair.

Reply:
Thank you